### PR TITLE
[test]:improve coverage of containerexec_connection_test.go in cloud/…

### PR DESCRIPTION
**Pull request by Dhiren-Mhatre**
/kind test

**What this PR does / why we need it**:
This PR adds additional unit tests for the ContainerExecConnection in the cloudstream package to increase test coverage. Test coverage has been improved from minimal coverage to 82%, covering various edge cases and logging paths.

Key improvements include:
- Added tests for error handling in the Serve method
- Coverage for EOF and read error paths
- Tests for context cancellation behavior
- Coverage for retry logic in sendCloseMessage

**Which issue(s) this PR fixes**:
Part of #6186 (Enhance KubeEdge testing coverage)

**Special notes for your reviewer**:
The PR focuses solely on improving test coverage without modifying the implementation code. Custom mocks were created to test various edge cases.

**Does this PR introduce a user-facing change?**:
```release-note
NONE

### CODE REVIEW

File: cloud/pkg/cloudstream/containerexec_connection_test.go [hunk 1]
```diff
@@ -164,3 +167,175 @@ func TestSendConnection_Exec(t *testing.T) {
...
+		MessageID:    1,
+		edgePeerStop: make(chan struct{}),
+		closeChan:    make(chan bool),
+	}
+
+	close(execConn.closeChan)
+
+	execConn.SetEdgePeerDone()
+
+	assert.True(true)
```
Comment (wbc6080) [inline cloud/pkg/cloudstream/containerexec_connection_test.go:309]: It simply closes the `closeChan` channel and then calls the `SetEdgePeerDone` method without checking whether the `edgePeerStop` channel receives the signal, so the correctness of the `SetEdgePeerDone` method cannot be verified.

I think you can start a goroutine to try to receive a signal from the `edgePeerStop` channel to verify that the method is correct like:
```
close(execConn.closeChan)

received := make(chan bool, 1)
    go func() {
        select {
        case <-execConn.edgePeerStop:
            received <- true
        default:
            received <- false
        }
    }()

execConn.SetEdgePeerDone()

```

Then use assert to determine the content of received chan

Comment (Dhiren-Mhatre) [reply cloud/pkg/cloudstream/containerexec_connection_test.go:312]: Thanks for the review and suggestion. I've applied your recommended approach using a goroutine to monitor the edgePeerStop channel and confirm it receives the signal.

```
func TestSetEdgePeerDone_Exec(t *testing.T) {
	assert := assert.New(t)

	execConn := &ContainerExecConnection{
		MessageID:    1,
		edgePeerStop: make(chan struct{}),
		closeChan:    make(chan bool),
	}

	received := make(chan bool, 1)
	go func() {
		select {
		case <-execConn.edgePeerStop:
			received <- true
		case <-time.After(100 * time.Millisecond):
			received <- false
		}
	}()

	execConn.SetEdgePeerDone()
	
	assert.True(<-received, "edgePeerStop channel should receive a signal")
}

```

Comment (wbc6080) [reply cloud/pkg/cloudstream/containerexec_connection_test.go:312]: Good job! And I think L299-L312 also need to be changed like this :)

Comment (Dhiren-Mhatre) [reply cloud/pkg/cloudstream/containerexec_connection_test.go]: Thank you for the follow-up!

**Comment by wbc6080** [comment]
lgtm
cc @fisherxu @Shelley-BaoYue @WillardHu
